src/login/logind-button.c

```diff
@@ -378,15 +483,20 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
 }
 
 static int button_suitable(int fd) {
-        unsigned long types[CONST_MAX(EV_KEY, EV_SW)/ULONG_BITS+1];
+        unsigned long types[max_of_n(2, EV_KEY, EV_SW)/ULONG_BITS+1];
 
         assert(fd >= 0);
 
         if (ioctl(fd, EVIOCGBIT(EV_SYN, sizeof types), types) < 0)
                 return -errno;
 
         if (bitset_get(types, EV_KEY)) {
-                unsigned long keys[CONST_MAX5(KEY_POWER, KEY_POWER2, KEY_SLEEP, KEY_SUSPEND, KEY_RESTART)/ULONG_BITS+1];
+                unsigned long keys[max_of_n(14, KEY_POWER, KEY_POWER2, KEY_SLEEP, KEY_SUSPEND, KEY_RESTART,
+                                            KEY_LEFTSHIFT, KEY_RIGHTSHIFT,
+                                            KEY_LEFTALT, KEY_RIGHTALT,
+                                            KEY_LEFTMETA, KEY_RIGHTMETA,
+                                            KEY_LEFTALT, KEY_RIGHTALT,
+                                            KEY_ESC)/ULONG_BITS+1];
```

Comment:
you are turning this into a VLA since you call a function here. that's ugly as fuck, given that the value actually isn't variable at all, but constant.

Please just define some macro somewhere KEY_MAX_INTERESTED that you define to the largest of the keys we care about, just by looking at input.h locally. And then use that constant value everywhere.

And then please in some loop where we iterate through all keys (for example in the loop I suggested below around bitset_get()) validate that the constant is accurate, by adding an assert() that checks that every KEY_xyz constant we bother with *actually* is below that macro.

Reply:
So KEY_MAX_INTERESTED is just the one with the highest value that I find manually, correct?

Reply:
@poettering I am sorry, I still have to get to this, but I feel like I am kind of hard-coding two different things, now that's OK if that's what I have to do, but what if I enforce in the const array that the values have to be in descending order? that way the first element serves as KEY_MAX_INTERESTED
   
In my mind that would kind of self-document itself a little bit, so that years down the line, no one would have to remember to update KEY_MAX_INTERESTED _and_ the array. Is this a bad idea?

Reply:
I decided against doing the weird array sorting stuff.
I seem to be missing something that is causing the docs to fail to build

Reply:
So I finally was able to test a makeosi built image, it seems the keyboard handling works! I see all the

```
Mar 22 02:26:15 localhost systemd-logind[392]: Secure Attention Key sequence pressed short.
Mar 22 02:26:15 localhost systemd-logind[392]: Secure Attention Key sequence pressed.
```
in the journal after running `journalctl --unit systemd-logind.service`

...how do I make sure the dbus bit is working though? Doesn't show up when I `busctl monitor --system`
Thanks

Reply:
Figured it out, I forgot the seat path in `sd_bus_emit_signal`